mutmut/__init__.py

```diff
@@ -572,19 +582,6 @@ def mutate_list_of_nodes(node, context):
 
         mutate_node(child_node, context=context)
 
-        # this is just an optimization to stop early
-        if context.number_of_performed_mutations and context.mutation_id != ALL:
-            return
```

Comment:
Are you sure you want to get rid of this early return?

Reply:
Restored this early return: it previously was haphazardly removed! 😬